core/App/screens/CredentialDetails.tsx

```diff
@@ -83,27 +136,40 @@ const CredentialDetails: React.FC<CredentialDetailsProps> = ({ navigation, route
   }, [])
 
   useEffect(() => {
-    if (!credential) {
-      return
-    }
-    const indyCredentialFormat = credential.credentials.find((c) => c.credentialRecordType === 'indy')
-    if (!indyCredentialFormat) {
+    if (!(credential && isValidIndyCredential(credential))) {
       return
     }
+
     credential.revocationNotification == undefined ? setIsRevoked(false) : setIsRevoked(true)
     if (isRevoked && credential?.revocationNotification?.revocationDate) {
       const date = new Date(credential.revocationNotification.revocationDate)
       setRevocationDate(date.toLocaleDateString(i18n.language, dateFormatOptions))
     }
-    OCABundle.getCredentialPresentationFields(credential as CredentialExchangeRecord, i18n.language).then((fields) =>
-      setFields(fields)
-    )
+
+    const resolveBundle = async () => {
+      const bundle = await OCABundle.resolve(credential)
+      const defaultBundle = await OCABundle.resolveDefaultBundle(credential)
+      return { bundle, defaultBundle }
+    }
+
+    const resolvePresentationFields = async () => {
+      const fields = await OCABundle.getCredentialPresentationFields(credential, i18n.language)
+      return { fields }
+    }
+
+    Promise.all([resolveBundle(), resolvePresentationFields()]).then(([{ bundle, defaultBundle }, { fields }]) => {
+      const overlayBundle = bundle || defaultBundle
+      const metaOverlay = overlayBundle?.getMetaOverlay(i18n.language)
+      const cardLayoutOverlay = overlayBundle?.getCardLayoutOverlay<CardLayoutOverlay>(CardOverlayType.CARD_LAYOUT_10)
```

Comment:
Not required in this PR but I think we favour `CardLayout10` or `CardLayout_10` in TypeScirpt.

Reply:
This is more in alignment with the define OCA types. Otherwise my OCD spots the differences in naming 😅. I can use an alias here.

Reply:
I think if we rename the OCA types I can just remove the underscores altogether.